lib/shared/src/completions/types.ts

```diff
@@ -1,16 +1,19 @@
 import type * as vscode from 'vscode'
 import type { URI } from 'vscode-uri'
 
+export type AutocompleteContextSnippetMetadataFields = Record<string, number | string>
+
 interface AutocompleteContextSnippetMetadata {
     /**
      * This field is relevant for user action context sources such as `recent-edit`, `recent-copy` and `recent-viewport`.
      * It indicates the time in milliseconds since the action was performed (eg: time Since the last edit).
      */
     timeSinceActionMs?: number
     /**
-     * The diffing strategy used by the `recent-edits-retriever` to generate the diff.
+     * Additional metadata fields that can be used to store arbitrary key-value pairs.
+     * The values can be either numbers or strings.
      */
-    recentEditsRetrieverDiffStrategy?: string
+    diffStrategyMetadata?: AutocompleteContextSnippetMetadataFields
```

Comment:
Does it make sense to name the type in the same way? 
```suggestion
    diffStrategyMetadata?: AutocompleteDiffStrategyMetadata
```

Reply:
Thanks for noticing,
Since This field is intended to capture additional metadata from the context retrievers and not limited by diff retriever. Changed the name to `retrieverMetadata`